File: backEnd/UnitedStoneTrustBank/src/main/java/com/ninos/audit_dashboard/service/AuditorService.java
package com.ninos.audit_dashboard.service;

import com.ninos.account.dto.AccountDTO;
import com.ninos.auth_users.dto.UserDTO;
import com.ninos.transaction.dto.TransactionDTO;
import java.util.List;
import java.util.Map;
import java.util.Optional;


public interface AuditorService {

    Map<String, Long> getSystemTotal();

    Optional<UserDTO> findUserByEmail(String email);

    Optional<AccountDTO> findAccountDetailsByAccountNumber(String accountNumber);

    List<TransactionDTO> findTransactionsByAccountNumber(String accountNumber);

    Optional<TransactionDTO> findTransactionById(Long transactionId);


}

File: frontEnd/unitedstonetrustbank-ui/src/services/auditorService.js
import api from "./api";

const AuditorService = {
    getSystemTotals: () => api.get("/audit/totals"),

    findUserByEmail: (email) => api.get(`/audit/users?email=${email}`),

    findAccountByAccountNumber: (accountNumber) =>
        api.get(`/audit/accounts?accountNumber=${accountNumber}`),

    findTransactionsByAccountNumber: (accountNumber) =>
        api.get(`/audit/transactions/by-account?accountNumber=${accountNumber}`),

    findTransactionById: (transactionId) =>
        api.get(`/audit/transactions/by-id?transactionId=${transactionId}`),
};

export default AuditorService;
